src/main/java/sg/ncl/MainController.java

```diff
@@ -1579,13 +1586,29 @@ public String checkJoinTeamInfo(
     //--------------------------Experiment Page--------------------------
 
     @RequestMapping(value = "/experiments", method = RequestMethod.GET)
-    public String experiments(Model model, HttpSession session) {
+    public String experiments(Model model, HttpSession session)throws WebServiceRuntimeException {
 //        long start = System.currentTimeMillis();
         List<Experiment2> experimentList = new ArrayList<>();
         Map<Long, Realization> realizationMap = new HashMap<>();
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        ResponseEntity response = restTemplate.exchange(properties.getDeterUid(session.getAttribute("id").toString()), HttpMethod.GET, request, String.class);
+
+        String responseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                log.error("No user to get experiment: {}", session.getAttribute("id"));
+                MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+                model.addAttribute(DETER_UID, CONNECTION_ERROR);
+            } else {
+                log.info("Show the deter user id: {}", responseBody);
+                model.addAttribute(DETER_UID, responseBody);
+            }
+        } catch (IOException e) {
+            throw new WebServiceRuntimeException(e.getMessage());
+        }
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)